### PR TITLE
WT-7984 Fix a bug that could cause a checkpoint to omit a page of data (v4.2 backport)

### PR DESCRIPTION
**Pull request by mm-ng**

### CODE REVIEW

**Review by lukech — APPROVED** [review]
Please make sure to run "git cherry-pick" with the `-x` command option to retain the original commit info from which it's cherry-picked.

File: src/reconcile/rec_write.c [hunk 1]
```diff
@@ -206,17 +209,17 @@ __reconcile(WT_SESSION_IMPL *session, WT_REF *ref, WT_SALVAGE_COOKIE *salvage, u
...
         __rec_write_page_status(session, r);
-    else
+    else {
+        /* Make sure that reconciliation doesn't free the page that has been written to disk. */
+        WT_ASSERT(session, addr == NULL || ref->addr != NULL);
         WT_TRET(__rec_write_wrapup_err(session, r, page));
-
-    /*
-     * If reconciliation completes successfully, save the stable timestamp.
-     */
```
Comment (kommiharibabu) [inline src/reconcile/rec_write.c:217]: Why this change is removed? The code needs to be kept.

File: src/reconcile/rec_write.c [hunk 2]
```diff
@@ -2182,6 +2185,37 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
...
+     */
+    WT_RET(__wt_ovfl_track_wrapup(session, page));
+
+    /*
+     * If using the history store table eviction path and we found updates that weren't globally
+     * visible when reconciling this page, copy them into the database's history store. This can
+     * fail, so try before clearing the page's previous reconciliation state.
+     */
+    if (F_ISSET(r, WT_REC_HS))
+        WT_RET(__rec_hs_wrapup(session, r));
```
Comment (kommiharibabu) [inline src/reconcile/rec_write.c:2209]: The HS changes need to be removed. I think they got introduced by mistake.

**Review by kommiharibabu — CHANGES_REQUESTED** [review]

File: src/reconcile/rec_write.c (hunk 2 above)
Comment (mm-ng) [reply src/reconcile/rec_write.c:2209]: Yes, I think something went wrong with the cherry-pick, I will fix these.
